internal/output/table.go

```diff
@@ -156,3 +170,84 @@ func MaxSeverity(group models.GroupInfo, pkg models.PackageVulns) string {
 
 	return fmt.Sprintf("%v", maxSeverity)
 }
+
+func licenseTableBuilder(outputTable table.Writer, vulnResult *models.VulnerabilityResults) table.Writer {
+	licenseConfig := vulnResult.ExperimentalAnalysisConfig.Licenses
+	if !licenseConfig.Enabled {
+		return outputTable
+	}
+	if len(licenseConfig.Allowlist) == 0 {
+		return licenseSummaryTableBuilder(outputTable, vulnResult)
+	}
+
+	return licenseViolationsTableBuilder(outputTable, vulnResult)
+}
+
+func licenseSummaryTableBuilder(outputTable table.Writer, vulnResult *models.VulnerabilityResults) table.Writer {
+	counts := make(map[models.License]int)
+	for _, pkgSource := range vulnResult.Results {
+		for _, pkg := range pkgSource.Packages {
+			for _, l := range pkg.Licenses {
+				counts[l] += 1
+			}
+		}
+	}
+	if len(counts) == 0 {
+		// No packages found.
+		return outputTable
+	}
+	licenses := maps.Keys(counts)
+	// Sort the license count in descending count order with the UNKNOWN
+	// license last.
+	sort.Slice(licenses, func(i, j int) bool {
+		if licenses[i] == "UNKNOWN" {
+			return false
+		}
+		if licenses[j] == "UNKNOWN" {
+			return true
+		}
+		if counts[licenses[i]] == counts[licenses[j]] {
+			return licenses[i] < licenses[j]
+		}
+
+		return counts[licenses[i]] > counts[licenses[j]]
+	})
+	outputTable.AppendHeader(table.Row{"License", "No. of package versions"})
+	for _, license := range licenses {
+		outputTable.AppendRow(table.Row{license, counts[license]})
+	}
+
+	return outputTable
+}
+
+func licenseViolationsTableBuilder(outputTable table.Writer, vulnResult *models.VulnerabilityResults) table.Writer {
+	outputTable.AppendHeader(table.Row{"License Violation", "Ecosystem", "Package", "Version", "Source"})
+	workingDir, err := os.Getwd()
+	if err != nil {
+		log.Panicf("can't get working dir: %v", err)
+	}
+	for _, pkgSource := range vulnResult.Results {
+		for _, pkg := range pkgSource.Packages {
+			if len(pkg.LicenseViolations) == 0 {
+				continue
+			}
+			violations := make([]string, len(pkg.LicenseViolations))
+			for i, l := range pkg.LicenseViolations {
+				violations[i] = string(l)
+			}
+			sourcePath, err := filepath.Rel(workingDir, pkgSource.Source.Path)
+			if err == nil { // Simplify the path if possible
+				pkgSource.Source.Path = sourcePath
```

Comment:
~~Does this change the JSON output to also be a relative path?~~

Actually probably not, since the Source field is not a pointer.

Reply:
I think theoretically this could change the JSON output, as vulnResult is a pointer. But to produce that error, it would need to be possible to produce JSON and table output in the same osv-scanner invocation, and I don't think that's possible?

either way, I've changed this so that we don't modify vulnResult, I think that's bad practice. good catch.